ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client/InFutureCheck.hs

```diff
@@ -0,0 +1,144 @@
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Ouroboros.Consensus.MiniProtocol.ChainSync.Client.InFutureCheck (
```

Comment:
@njd42 Please confirm this logic matches what you and I chatted about a few weeks ago. Thanks!

Reply:
@njd42 It's being merged now, but I'd still appreciate your thoughts on this module. Thanks.